### PR TITLE
Feature/add soy view

### PR DESCRIPTION
**Pull request by caiocacastro**
All working fine now. Can you please evaluate the code, my man?

### CODE REVIEW

File: detection/views.py [hunk 1]
```diff
@@ -1,15 +1,14 @@
-
 import logging
 from datetime import datetime
 
 from requests.models import Response
 from rest_framework import generics, response, status
 
-from .models import BasicElasticStructure
-from .utils import UtilFunctions
+from elastic.models import Structure as ElasticStructure
```
Comment (dagnaldo) [inline detection/views.py]: change import to `from elastic import models, utils`

File: elastic/utils.py [hunk 2]
```diff
@@ -17,8 +17,9 @@
 from django.conf import settings
 from rest_framework import status
 
-from .serializers import DetectionSerializer
-from .models import Detection, BasicElasticStructure
+from detection.serializers import DetectionSerializer
```
Comment (dagnaldo) [inline elastic/utils.py]: change imports
```
from detection import serializers, models
from elastic import models as elastic_models
```

File: soy/views.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+            insertion_errors = self.util_class.send_bulk_list(
+                soy_series, es_structure
+            )
+
+            if insertion_errors:
+                response_data = {
+                    'msg': 'Some data were not inserted',
+                    'number_of_errors': len(insertion_errors),
+                    'errors_id_list': 
+                        [error['id'] for error in insertion_errors],
```
Comment (dagnaldo) [inline soy/views.py:62]: `id` attribute exists on data?

File: detection/serializers.py [hunk 4]
```diff
@@ -25,8 +25,6 @@ class Meta:
             'dt_t_um',
             'nu_area_km2',
             'nu_area_ha',
-            'latitude',
```
Comment (dagnaldo) [inline detection/serializers.py]: Did you remove latitude and longitude from detection?

**Review by dagnaldo — CHANGES_REQUESTED** [review]

File: elastic/utils.py (hunk 2 above)
Comment (caiocacastro) [reply elastic/utils.py]: Done 745678d

File: detection/views.py (hunk 1 above)
Comment (caiocacastro) [reply detection/views.py]: Done 745678d

File: soy/views.py (hunk 3 above)
Comment (caiocacastro) [reply soy/views.py:62]: Done 745678d

File: detection/serializers.py (hunk 4 above)
Comment (caiocacastro) [reply detection/serializers.py]: Done 745678d

File: docker-compose.yaml [hunk 5]
```diff
@@ -53,6 +53,8 @@ services:
       DJANGO_LOG_LEVEL: 'INFO'
       DEBUG: 'true'
       ALLOWED_HOSTS: '["*"]'
+      DETECTION_TEST_URL: https://storage.googleapis.com/xskylab-catalog/SUPPORT/TESTS/FUNAI/
```
Comment (dagnaldo) [inline docker-compose.yaml]: Change URL to work direct with file.

File: docker-compose.yaml [hunk 6]
```diff
@@ -53,6 +53,8 @@ services:
...
+      SOY_TEST_URL: https://storage.googleapis.com/xskylab-catalog/SUPPORT/TESTS/SOY/
```
Comment (dagnaldo) [inline docker-compose.yaml]: Change URL to work direct with file.

File: elastic/tests/test_structure.py [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import os
+
+from django.test import TestCase
+
+
+class TestElastic(TestCase):
```
Comment (dagnaldo) [inline elastic/tests/test_structure.py:6]: Missing blank line between class and method.

File: service_load_data_elasticsearch/settings.py [hunk 8]
```diff
@@ -168,9 +169,13 @@
...
-BULK_SIZE_REQUEST = os.getenv("BULK_SIZE_REQUEST", 1000)
+BULK_SIZE_REQUEST = os.getenv('BULK_SIZE_REQUEST', 1000)
 ELASTICSEARCH_URL = os.getenv(
-    "ELASTICSEARCH_URL", "http://es.xskylab.com:9200")
+    'ELASTICSEARCH_URL', 'http://es.xskylab.com:9200')
 
 # Variable controlling SLL validation for es requests
-VERIFY_SSL = bool(distutils.util.strtobool(os.getenv("VERIFY_SSL", True)))
\ No newline at end of file
+VERIFY_SSL = bool(distutils.util.strtobool(os.getenv('VERIFY_SSL', 'true')))
```
Comment (dagnaldo) [inline service_load_data_elasticsearch/settings.py:177]: Very well, sir!

File: soy/tests/soy_test_error_file.txt [hunk 9]
```diff
@@ -0,0 +1,6 @@
+{"index": {}}
```
Comment (dagnaldo) [inline soy/tests/soy_test_error_file.txt]: Why does this file exists?

File: soy/tests/soy_test_file.txt [hunk 10]
```diff
@@ -0,0 +1,6 @@
+{"index": {}}
```
Comment (dagnaldo) [inline soy/tests/soy_test_file.txt]: Why does this file exists?

File: soy/tests/test_views.py [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+from django.contrib.staticfiles import finders
+from rest_framework.test import APIClient
+from rest_framework import status
+
+from soy import utils
+
+from .recipes import Recipes
+
+
+class TestSoy(TestCase):
```
Comment (dagnaldo) [inline soy/tests/test_views.py:15]: Missing blank line

File: soy/utils.py [hunk 12]
```diff
@@ -0,0 +1,112 @@
...
+
+from datetime import datetime
+
+from elastic import utils
+
+
+logger = logging.getLogger('django')
+
+
+class Utils(utils.Utils):
```
Comment (dagnaldo) [inline soy/utils.py:12]: Missing blank line.

**Review by dagnaldo — CHANGES_REQUESTED** [review]

File: docker-compose.yaml (hunk 5 above)
Comment (caiocacastro) [reply docker-compose.yaml]: url straigth to the file? It will need 4 urls, then.. Are you sure?

File: docker-compose.yaml (hunk 6 above)
Comment (dagnaldo) [reply docker-compose.yaml]: Yes, please.

Comment (caiocacastro) [reply docker-compose.yaml]: Done, fellow friend bbeec4c

File: elastic/tests/test_structure.py (hunk 7 above)
Comment (caiocacastro) [reply elastic/tests/test_structure.py:6]: Done, fellow compatriot bbeec4c

File: service_load_data_elasticsearch/settings.py (hunk 8 above)
Comment (caiocacastro) [reply service_load_data_elasticsearch/settings.py:177]: Thanks frienderino!

File: soy/tests/soy_test_error_file.txt (hunk 9 above)
Comment (caiocacastro) [reply soy/tests/soy_test_error_file.txt]: Removed! bbeec4c

File: soy/tests/soy_test_file.txt (hunk 10 above)
Comment (caiocacastro) [reply soy/tests/soy_test_file.txt]: Removed! bbeec4c

File: soy/tests/test_views.py (hunk 11 above)
Comment (caiocacastro) [reply soy/tests/test_views.py:15]: Added! bbeec4c

File: soy/utils.py (hunk 12 above)
Comment (caiocacastro) [reply soy/utils.py:12]: A......nd another added! bbeec4c

**Review by dagnaldo — APPROVED** [review]
All chekcs passed.
I will test it now, btw, it's approved.

**Pull request merged**